src/engine.ml

```diff
@@ -101,20 +105,28 @@ let hmacs config =
   match Config.find Tls_auth config with
   | None -> Error (`Msg "no tls auth payload in config")
   | Some (direction, _, hmac1, _, hmac2) ->
+      let hmac_len =
+        Option.value ~default:`SHA1 (Config.find Auth config)
+        |> Mirage_crypto.Hash.digest_size
+      in
```

Comment:
I'd propose to move ``Auth `SHA1`` into the default configuration, and avoid the `Option.value ~default` (that appears thrice in this PR). WDYT?

Reply:
Yea, that makes sense. That's what openvpn does too in fact!